apps/snitch_core/lib/core/data/model/product.ex

```diff
@@ -174,6 +174,8 @@ defmodule Snitch.Data.Model.Product do
   def delete(id) do
     with %Product{} = product <- get(id),
          changeset <- Product.delete_changeset(product) do
+      product = product |> Repo.preload(:images)
```

Comment:
also add a call to elasticsearch to remove the product from index like this 
`ESProductStore.update_product_to_es(id, :delete)`